### PR TITLE
Add missing cleanup for block types.

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Proposed Changes

Cleans up block definitions in test teardown

### Reason for Changes

Reduces warnings thrown by tests

### CODE REVIEW

File: tests/mocha/block_test.js [hunk 1]
```diff
@@ -324,16 +325,6 @@ suite('Blocks', function() {
...
-          "args0": [
-            {
-              "type": "input_statement",
-              "name": "STATEMENT"
-            }
-          ]
-        },
       ]);
     });
     teardown(function() {
```
Comment (rachel-fenichel) [inline tests/mocha/block_test.js:330]: This teardown needs to not remove statement block.

File: tests/mocha/block_test.js [hunk 2]
```diff
@@ -1030,13 +1021,6 @@ suite('Blocks', function() {
...
-      Blockly.defineBlocksWithJsonArray([
-        {
-          "type": "empty_block",
-          "message0": "",
-          "args0": []
-        },
-      ]);
       this.eventSpy = sinon.spy(Blockly.Events, 'fire');
     });
     teardown(function() {
```
Comment (rachel-fenichel) [inline tests/mocha/block_test.js:1026]: Same here: the teardown should not remove the empty block

File: tests/mocha/navigation_test.js [hunk 3]
```diff
@@ -367,6 +367,9 @@ suite('Navigation', function() {
         }
       };
     });
+    teardown(function() {
+      delete Blockly.Blocks['basic_block'];
```
Comment (rachel-fenichel) [inline tests/mocha/navigation_test.js:371]: Add: `this.workspace.dispose();`

The teardown should have been doing it before, but wasn't.

File: tests/mocha/block_test.js (hunk 1 above)
Comment (moniika) [reply tests/mocha/block_test.js:330]: Done

File: tests/mocha/block_test.js (hunk 2 above)
Comment (moniika) [reply tests/mocha/block_test.js:1026]: Done

File: tests/mocha/navigation_test.js (hunk 3 above)
Comment (moniika) [reply tests/mocha/navigation_test.js:371]: Done

**Pull request merged**